packages/nimble-components/src/wafer-map/modules/rendering.ts

```diff
@@ -10,13 +10,28 @@ export class RenderingModule {
 
     public constructor(private readonly wafermap: WaferMap) {}
 
-    public drawWafer(): void {
+    public updateSortedDies(): void {
+        this.dies = this.wafermap.dataManager.diesRenderInfo.sort((a, b) => {
+            if (a.fillStyle > b.fillStyle) {
+                return 1;
+            }
+            if (b.fillStyle > a.fillStyle) {
+                return -1;
+            }
+
+            return 0;
+        });
+        this.updateDrawnWafer();
+    }
+
+    public updateDrawnWafer(): void {
         this.wafermap.canvasContext.save();
         this.clearCanvas();
         this.scaleCanvas();
-        this.sortDies();
         this.renderDies();
-        this.renderText();
+        if (!this.wafermap.dieLabelsHidden) {

```

Comment:
nit: Any reason not to put this logic in the `renderText` method? I don't feel too strongly about this, but sometimes if each step has conditions associated with it (doesn't seem to the case here for now), then avoiding a bunch of `if` statements in a row can help maintain clarity of intent.